src/clims/api/endpoints/process_assignments.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import absolute_import
+
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint, SessionAuthentication
+
+
+class ProcessAssignmentsEndpoint(Endpoint):
```

Comment:
Yes I understand that, thanks for bringing this up. :)

(Please note first that this is in the middle of refactoring it to always be named process, never workflow, that explains the chaotic switches between names.)

This is a bit of a long explanation of my thinking here, as it also goes into how I suggest we think of naming in general. It would be great to get a consensus on what's acceptable in the project so I'd like to ping the team too: @commonlims/core.

**Workflow vs. Process**

So why not workflow instead of process? It's a tough decision in my mind, because in Camunda and in the BPMN standard (check out https://www.omg.org/spec/BPMN/2.0/PDF, section Process) these are called processes. They sometimes call it Workflow, but informally. The executable workflow is called a Process.

If we start naming it differently there is a certain disconnect between what we call it and what Camunda calls it, which complicates things and creates work for us, because we need to create new names for some things (e.g. `process_definition_key` must be `workflow_definition_key`). Also, when debugging you have to remind yourself of this mapping all the time.

I thought it might be a good middle ground to call it "workflows" at the top layer, but "process" when communicating with Camunda, and this is what I started doing. But that also has the downside in that it gets confusing when you get further down the stack and everything is all of a sudden called process.

I have a tendency to think, having worked a bunch with the workflow feature for a while now, that it's simpler to just call it the same as Camunda calls it and then document clearly what processes are in the context of Common LIMS.

**Naming in general**

I think it's fine to use generic words in programming, as long as it's clear to everyone what the generic word means in the scope. The team discussed a bunch until we came up with the word `Substance` for what's in a container, and I think that was great, but it's definitely still a generic word. Problem is that all other words we thought of were also generic, only some more than others (e.g. Item).

An example of a highly generic word we use regularly is when we use the the word "print" to print to the console in Python. It's a generic word in the sense that it can also mean "print to a printer" (it even means that in javascript), but every Python programmer knows that it doesn't mean that in the context of a Python program. I would argue that it's OK that they thought it meant "print to a printer" their first day learning Python, as long as they know better the second day :)

**Conclusion**

So from my perspective it's OK to call workflows processes, because we're in the context of a workflow engine. Also, because it's "process assignment" it's hopefully clear that it must be assignment to a workflow process, because we don't have the concept of assigning to e.g. an operating system process. And this will in the end be clear in our documentation.

**Follow up**

I suggest a guideline which the whole team (ping @commonlims/core) might want to chime in on: If we think that a term becomes confusing after reading the documentation (of the end product), we should change it, otherwise it's ok.

And of course, if you think otherwise, please say so!